### PR TITLE
Skip lis tests when lis driver doesn't support install on current distro

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: lisa/features/gpu.py [hunk 1]
```diff
@@ -122,8 +122,7 @@ def install_compute_sdk(self, version: str = "") -> None:
         self._install_gpu_dep()
         try:
             # install LIS driver if required and not already installed.
-            if LisDriver.can_install:
-                self._node.tools[LisDriver]
+            self._node.tools[LisDriver]
         except Exception as identifier:
             self._log.debug(
                 "LisDriver is not installed. It might not be required. " f"{identifier}"
```
Comment (squirrelsc) [inline lisa/features/gpu.py]: remove the `" f"`. It doesn't need in one line.

File: lisa/sut_orchestrator/azure/tools.py [hunk 2]
```diff
@@ -187,12 +192,14 @@ def command(self) -> str:
 
     @property
     def can_install(self) -> bool:
+        return True
+
+    def is_support(self) -> bool:
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/tools.py]: Why not use `can_install`?

File: lisa/features/gpu.py [hunk 3]
```diff
@@ -122,8 +122,7 @@ def install_compute_sdk(self, version: str = "") -> None:
         self._install_gpu_dep()
         try:
             # install LIS driver if required and not already installed.
-            if LisDriver.can_install:
```
Comment (squirrelsc) [inline lisa/features/gpu.py:125]: The `can_install` is called by LISA, it doesn't need to be called explicitly.

File: lisa/sut_orchestrator/azure/tools.py (hunk 2 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/tools.py]: it returns an object, always return true

File: lisa/features/gpu.py (hunk 3 above)
Comment (LiliDeng) [reply lisa/features/gpu.py:125]: hmm, yes, let me retry

File: lisa/features/gpu.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/features/gpu.py]: removed

File: lisa/features/gpu.py (hunk 3 above)
Comment (LiliDeng) [reply lisa/features/gpu.py:125]: updated

**Comment by LiliDeng** [comment]
either we skip test here, or add a failure type in db

File: lisa/sut_orchestrator/azure/tools.py [hunk 4]
```diff
@@ -192,8 +197,10 @@ def can_install(self) -> bool:
             and self.node.os.information.version < "7.8.0"
         ):
             return True
-
-        return False
+        raise SkippedException(
```
Comment (squirrelsc) [inline lisa/sut_orchestrator/azure/tools.py]: Don't skip test case in the tool, catch it and use Skipped in test case.

File: microsoft/testsuites/lis/lissuite.py [hunk 5]
```diff
@@ -121,97 +121,90 @@ def _verify_lis_preinstall_disk_size(
...
-                    )
-                    fallocate.create_file(
-                        single_partition_file_size, "/lib/modules/file.out"
-                    )
-                if boot_partition != root_partition:
-                    single_partition_file_size_boot = (
-                        boot_part_avail_space - boot_part_required_space
+                fallocate.create_file(single_partition_file_size_boot, "/boot/file.out")
+            result = lisdriver.install_from_iso()
+            if test_type == "negative":
```
Comment (squirrelsc) [inline microsoft/testsuites/lis/lissuite.py:198]: `if test_type == "negative" and result.exit_code == 0:`

File: microsoft/testsuites/lis/lissuite.py [hunk 6]
```diff
@@ -121,97 +121,90 @@ def _verify_lis_preinstall_disk_size(
...
-                self._clean_up_files(node)
-                if test_type == "positive":
-                    lisdriver.uninstall_from_iso()
-            else:
-                raise SkippedException(
-                    f"current distro {node.os.name} doesn't support install LIS"
-                )
+                else:
+                    log.debug(f"fail to install lis for reason {result.stdout}")
+            self._clean_up_files(node)
```
Comment (squirrelsc) [inline microsoft/testsuites/lis/lissuite.py]: This line may not be called, if exception raised above. call it in after_case

File: lisa/sut_orchestrator/azure/tools.py (hunk 4 above)
Comment (LiliDeng) [reply lisa/sut_orchestrator/azure/tools.py]: updated

File: microsoft/testsuites/lis/lissuite.py (hunk 6 above)
Comment (LiliDeng) [reply microsoft/testsuites/lis/lissuite.py]: updated

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**